ui/models/push.js

```diff
@@ -151,24 +154,44 @@ export default class PushModel {
   }
 
   static async getDecisionTaskId(pushId, notify) {
-    const taskIdMap = await PushModel.getDecisionTaskMap([pushId], notify);
+    const taskId = decisionTaskIdCache[pushId];
 
-    return taskIdMap[pushId];
+    if (taskId) {
+      return taskId;
+    }
+    Object.assign(
+      decisionTaskIdCache,
+      await PushModel.getDecisionTaskMap([pushId], notify),
+    );
+    return decisionTaskIdCache[pushId];
   }
 
   static async getDecisionTaskMap(pushIds, notify) {
+    // If any Push ids are not yet in the cache, then fetch them.
+    // Otherwise just return the map since it has everything
+    // that's needed.
+    const cachedMap = pick(decisionTaskIdCache, pushIds);
+    const missedIds = pushIds.filter(id => !cachedMap[id]);
+
+    if (!Object.keys(missedIds).length) {
+      return cachedMap;
+    }
+
+    // Some Push ids were not yet in the cache, so must fetch them.
     const { data, failureStatus } = await getData(
-      getProjectUrl(`${pushEndpoint}decisiontask/?push_ids=${pushIds}`),
+      getProjectUrl(`${pushEndpoint}decisiontask/?push_ids=${missedIds}`),
     );
 
     if (failureStatus) {
-      const msg = `Error getting Gecko Decision Task Ids: ${failureStatus}: ${data}`;
+      const msg = `Error getting Gecko Decision Tasks for ids: ${pushIds}: ${failureStatus}: ${data}`;
```

Comment:
Should this variable be `missedIds`, not `pushIds`?

Reply:
Right again...  :)  Fixed.